homeassistant/components/ecobee/binary_sensor.py

```diff
@@ -52,6 +52,40 @@ def unique_id(self):
                     return f"{sensor['code']}-{self.device_class}"
                 return f"{sensor['id']}-{self.device_class}"
 
+    @property
+    def device_info(self):
+        """Return device information for this sensor."""
+        identifier = None
+        model = None
+        for sensor in self.data.ecobee.get_remote_sensors(self.index):
+            if sensor["name"] == self.sensor_name:
+                if "code" in sensor:
+                    identifier = sensor["code"]
+                    model = "ecobee Room Sensor"
+                else:
+                    thermostat = self.data.ecobee.get_thermostat(self.index)
+                    identifier = thermostat["identifier"]
+                    try:
+                        model = f"{ECOBEE_MODEL_TO_NAME[thermostat['modelNumber']]} Thermostat"
+                    except KeyError:
+                        _LOGGER.error(
+                            "Model name for ecobee thermostat %s not recognized. "
+                            "Please open an issue on GitHub and provide this information: "
+                            "Unrecognized model: %s",
+                            thermostat["name"],
+                            thermostat["modelNumber"],
+                        )
```

Comment:
This feels inelegant but the ecobee API documentation is not in sync with the new devices that are released (e.g. there is no info for ecobee4 model name/number) and will signal to the user to open an issue so that the proper name can be added.

Reply:
I think that this is fine. We need to build up that device database one way or another . Maybe create an issue whose link we can include here? That makes it easier to report.